### PR TITLE
[diagnostics] Mark attributes that we always intercept as External to save RAM.

### LINKED ISSUE #17870: [diagnostics]For attributes that we always intercept, we can mark them external to save RAM
#### Problem
<what's wrong or missing, please include any applicable:
 * expected behavior
 * actual behavior
 * steps to reproduce
 * system configuration
 * screenshots, images, logs, etc.
 >

 #### Proposed Solution
<suggested fix, suggested enhancement>

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
What is being fixed?  Examples:
* For attributes that we always intercept, we can mark them external to save RAM, since the source of truth of those attributes are on platform, fall back to read and write from/to the attribute store does not make sense in reality. I started this optimization from diagnostic clusters in this PR

* Fixes #17870

#### Change overview
Mark attributes that we always intercept as External to save RAM.

#### Testing
How was this tested? (at least one bullet point required)
* No logic change, the regression is covered by CI

### CODE REVIEW

**Review by bzbarsky-apple — COMMENTED** [review]
It would be better to just add these to the `attributeAccessInterfaceAttributes` list in `src/app/zap-templates/zcl/zcl.json`, instead of changing all the .zap files by hand...

**Review by tcarmelveilleux — APPROVED** [review]

**Review by mlepage-google — APPROVED** [review]

**Comment by yufengwangca** [comment]
> It would be better to just add these to the `attributeAccessInterfaceAttributes` list in `src/app/zap-templates/zcl/zcl.json`, instead of changing all the .zap files by hand...

Acked,  I found two attributes 'bssid' and 'SecurityType' in WiFi Network Diagnostics are not updated in zap, but as long as the UI and the db and the codegen is changed to External, it should be fine.

File: src/app/zap-templates/zcl/zcl.json [hunk 1]
```diff
@@ -133,6 +133,37 @@
             "TxErrCount",
             "CollisionCount",
             "OverrunCount"
+        ],
+        "General Diagnostics": [
+            "NetworkInterfaces",
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/zcl.json:138]: Fwiw, putting list-typed things in here does not hurt anything, but is not necessary either: lists are detected by type and always forced to be external anyway.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/zap-templates/zcl/zcl.json (hunk 1 above)
Comment (yufengwangca) [reply src/app/zap-templates/zcl/zcl.json:138]: Thanks!

**Comment by bzbarsky-apple** [comment]
@yufengwangca The unreviewed changes in https://github.com/project-chip/connectedhomeip/pull/17928/commits/698323e3d1cbc99508b5fc075762687c51c4f073 should not have been needed.  Why are they needed?

**Comment by yufengwangca** [comment]
> @yufengwangca The unreviewed changes in [698323e](https://github.com/project-chip/connectedhomeip/commit/698323e3d1cbc99508b5fc075762687c51c4f073) should not have been needed. Why are they needed?

Yes, that is the issue we discussed last night, I am not able to pass CI without this change, after I change the attribute to External, we need to call the platform APIs instead of attribute accessor to read and write the attributes

**Comment by bzbarsky-apple** [comment]
Ah, this is because of the changes in `emberAfSoftwareDiagnosticsClusterResetWatermarksCallback`, not because the of the attribute access interface bits per se?

Seems like the right thing to do might have been to just skip doing the `SetCurrentHeapHighWatermark` if `GetCurrentHeapUsed` returns "not implemented".... 

In fact, it's enirely valid for a node to not impleme the `GetCurrentHeapUsed` attribute but implement the `WTRMRK` feature, right?  In which case the implementation we had both before this PR and after this PR makes no sense....

**Pull request merged**